src/carbon_sendgrid_adapter.cr

```diff
@@ -138,6 +139,19 @@ class Carbon::SendGridAdapter < Carbon::Adapter
       ].compact
     end
 
+    private def attachments : Array(Hash(String, String))
+      files = [] of Hash(String, String)
+      email.attachments.map do |attachment|
+        case attachment
+        in AttachFile, ResourceFile
```

Comment:
I don't know what the difference between `AttachFile` and `ResourceFile` are, but I'm guessing it's related to SMTP and doesn't really matter in this case.